app/models/manageiq/providers/amazon/inventory/targets/ems_event_collection.rb

```diff
@@ -0,0 +1,186 @@
+        :arel     => ems.hardwares.joins(:vm_or_template).where(
+          :vms => {:ems_ref => manager_refs}),
+        :strategy => :find_missing_in_local_db))
+
+  end
+
+  def add_cloud_networks_inventory_collections(manager_refs)
+    return if manager_refs.blank?
+
+    add_inventory_collection(
+      cloud_networks_init_data(
+        :arel     => ems.network_manager.cloud_networks.where(:ems_ref => manager_refs),
+        :strategy => :find_missing_in_local_db))
+  end
+
+  def add_cloud_subnets_inventory_collections(manager_refs)
+    return if manager_refs.blank?
+
+    add_inventory_collection(
+      cloud_subnets_init_data(
+        :arel     => ems.network_manager.cloud_subnets.where(:ems_ref => manager_refs),
+        :strategy => :find_missing_in_local_db))
+  end
+
+  def add_security_groups_inventory_collections(manager_refs)
+    return if manager_refs.blank?
+
+    add_inventory_collection(
+      security_groups_init_data(
+        :arel     => ems.network_manager.security_groups.where(:ems_ref => manager_refs),
+        :strategy => :find_missing_in_local_db))
+    add_inventory_collection(
+      firewall_rules_init_data(
+        :arel     => ems.network_manager.firewall_rules.references(:security_groups).where(
+          :security_groups => {:ems_ref => manager_refs}),
+        :strategy => :find_missing_in_local_db))
+  end
+
+  def add_network_ports_inventory_collections(manager_refs)
+    return if manager_refs.blank?
+
+    add_inventory_collection(
+      network_ports_init_data(
+        :arel     => ems.network_manager.network_ports.where(:ems_ref => manager_refs),
+        :strategy => :find_missing_in_local_db))
+    add_inventory_collection(
+      cloud_subnet_network_ports_init_data(
+        :arel     => ems.network_manager.cloud_subnet_network_ports.references(:network_ports).where(
+          :network_ports => {:ems_ref => manager_refs}),
+        :strategy => :find_missing_in_local_db))
+  end
+
+  def add_floating_ips_inventory_collections(manager_refs)
+    return if manager_refs.blank?
+
+    add_inventory_collection(
+      floating_ips_init_data(
+        :arel     => ems.network_manager.floating_ips.where(:ems_ref => manager_refs),
+        :strategy => :find_missing_in_local_db))
+  end
+
+  def parse_ems_events!
+    uniq_checker = {}
+
+    target.all_related_ems_events(ems).each do |ems_event|
+      resource_type     = ems_event[:full_data]["configurationItem"]["resourceType"]
+      resource_id       = ems_event[:full_data]["configurationItem"]["resourceId"]
+      unique_checker_id = "#{resource_type}__#{resource_id}"
+
+      next if resource_type.blank? || resource_id.blank?
+      # We take only the last change and ignore the rest. Events are ordered so the last ones come first.
+      next if uniq_checker[unique_checker_id]
+
+      uniq_checker[unique_checker_id] = true
+
+      event_payload   = event_payload(ems_event)
+      collection_name = case resource_type
```

Comment:
Yes, we are going to add storage events, but I was planning to add them after we have at least the basic logic working.